cli/src/api/pcluster_api.py

```diff
@@ -80,34 +102,50 @@ def delete_cluster(cluster_name: str, region: str, keep_logs: bool = True):
             # retrieve cluster config and generate model
             cluster = Cluster(cluster_name)
             cluster.delete(keep_logs)
-            return ClusterInfo(cluster)
-        except (ClusterActionError, Exception) as e:
-            return ApiFailure(e)
+            return FullClusterInfo(cluster)
+        except Exception as e:
+            return ApiFailure(str(e))
 
     @staticmethod
     def describe_cluster(cluster_name: str, region: str):
         """Get cluster information."""
         try:
             if region:
                 os.environ["AWS_DEFAULT_REGION"] = region
-            return ClusterInfo(Cluster(cluster_name))
-        except (ClusterActionError, Exception) as e:
-            return ApiFailure(e)
+            return FullClusterInfo(Cluster(cluster_name))
```

Comment:
By looking at the API deign, the responses are different for different API calls. Here, `create cluster, describe_cluster` all return the same `FullClusterInfo` object. I am wondering, should we wrap the output here or in command modules or when we integrate with Lambda?

Reply:
I'm trying to build the classes looking forward to the new api design, but we can't address everything right now.
It will be part of the API tasks to adapt the outputs and return the right things.

For now I'm trying to give the user the same user-experience they had before this patch, then we'll refine it.